src/acl.rs

```diff
@@ -144,11 +144,26 @@ pub struct ACLEntry {
     pub string_sid: String,
 }
 
+#[derive(Debug)]
+enum ACLSource {
+    Path(String),
+    Handle(HANDLE)
+}
+
+impl ACLSource {
```

Comment:
Nitpick: instead of a custom `to_sd_source`, could this be an `impl From<ACLSource> for SDSource`?

Reply:
`From` consumes the input value and `SDSource::Path` is just a reference of `ACLSource::Path`. I added this two enums because `ACL` needs an owned `String` but `SecurityDescriptor` constructors take a `&str` reference. But maybe `SDSource` is not needed even and we can just pass `ACLSource` instead?

Reply:
That makes sense -- I forgot that `SDSource` borrowed its path. Let's leave this as-is then.